zkevm-circuits/src/evm_circuit/step.rs

```diff
@@ -80,7 +80,8 @@ pub enum ExecutionState {
     BLOCKHASH,
     BLOCKCTXU64,  // TIMESTAMP, NUMBER, GASLIMIT
     BLOCKCTXU160, // COINBASE
-    BLOCKCTXU256, // DIFFICULTY, BASEFEE
+    BLOCKCTXU256, // BASEFEE
+    DIFFICULTY,   // DIFFICULTY
```

Comment:
We can only introduce this state only if the `scroll` feature is set. Otherwise an soundness issue will arise. See the above comment.

Reply:
I suggest to change this to 
```
BLOCKCTXU256, // BASEFEE, (DIFFICULTY if not "scroll")
#[cfg(feature = "scroll")]
DIFFICULTY,
```

Reply:
Fixed in commit [a511ed4](https://github.com/scroll-tech/zkevm-circuits/pull/1033/commits/a511ed4157341a9555ef28297914000a935287c5).